hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/AbstractScheduleHandler.java

```diff
@@ -16,75 +16,361 @@
         try {
-            scheduledFunction = functionOf(scheduledTxn);
-        } catch (UnknownHederaFunctionality ex) {
-            throw new PreCheckException(SCHEDULED_TRANSACTION_NOT_IN_WHITELIST);
+            // note, payerAccount should never be null, but we're playing it safe here.
+            final AccountID payer = scheduleInState.payerAccountOrElse(context.payer());
+            final TransactionBody scheduledAsOrdinary = ScheduleUtility.asOrdinary(scheduleInState);
+            final TransactionKeys keyStructure = context.allKeysForTransaction(scheduledAsOrdinary, payer);
+            final Set<Key> scheduledRequiredKeys = getKeySetFromTransactionKeys(keyStructure);
+            final Set<Key> currentSignatories = setOfKeys(scheduleInState.signatories());
+            scheduledRequiredKeys.removeAll(currentSignatories);
+            final Set<Key> remainingRequiredKeys = scheduledRequiredKeys.parallelStream()
+                    .map(new ValidatedKeysMapping(context, currentSignatories))
+                    .flatMap(Set::stream)
+                    .collect(Collectors.toSet());
+            // This works ONLY because ValidatedKeysMapping observes the validation process and
+            // modifies currentSignatories to add any newly validated primitive keys.
+            return new ScheduleKeysResult(currentSignatories, remainingRequiredKeys);
+        } catch (final PreCheckException translated) {
+            throw new HandleException(translated.responseCode());
         }
+    }
 
-        if (!isSchedulable(scheduledFunction)) {
-            throw new PreCheckException(SCHEDULED_TRANSACTION_NOT_IN_WHITELIST);
+    /**
+     * Given an arbitrary {@link Iterable<Key>}, return a <strong>modifiable</strong> {@link Set<Key>} containing
+     * the same objects as the input.
+     * If there are any duplicates in the input, only one of each will be in the result.
+     * If there are any null values in the input, those values will be excluded from the result.
+     * Note: The return value of this method is always safe to be read and written by multiple threads
+     *     concurrently, and operates as close to lock-free as possible.
+     * Implementation Note: we use ConcurrentSkipListSet that orders the entries by hash code.  This means that
+     *   object order will be very different from a reasonable natural ordering, but it should (if hashCode is defined
+     *   properly) be consistent with equals, and have sufficient diffusion to work well with the random
+     *   functions inherent in the Skip List data structure. Order is not important, so this approach is reasonable.
+     *   The alternative is a CopyOnWriteArraySet, which has significant GC performance issues.
+     * @param keyCollection an Iterable of Key values.
+     * @return a {@link Set<Key>} containing the same contents as the input.  Duplicates and null values are excluded
+     * from this Set.  This Set is always a modifiable set..
```

Comment:
Is there significance to `set..` instead of `set.` (just one period)?

Reply:
accidental double type of the key.

Reply:
fixed.